### PR TITLE
Fix side effect bug for NodeFunctionGeneric, 0-1 arity and 0-1 result…

### PR DESCRIPTION
**Pull request by carlitosan**
…s nodes are restored to pre-bug-fix behavior

A previous bug for for node function generics introduced a bug for nodes that had either 0-1 results, or 0-1 inputs, by adding superfluous slots if the programmer had NOT named those slots manually. This bug would only appear by opening the graph in the editor, and then saving and running it. This change leaves the original bug fixed, while not breaking old graphs using these nodes.

Signed-off-by: carlitosan <82187351+carlitosan@users.noreply.github.com>

### CODE REVIEW

**Review by lsemp3d — APPROVED** [review]

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/NodeFunctionGeneric.h [hunk 1]
```diff
@@ -80,10 +80,14 @@ namespace ScriptCanvas
         using ResultType = FunctionTraits::result_type;\
         static const size_t s_numArgs = FunctionTraits::arity;\
         static const size_t s_numNames = SCRIPT_CANVAS_FUNCTION_VAR_ARGS(__VA_ARGS__);\
-        /*static const size_t s_numResults = ScriptCanvas::Internal::extended_tuple_size<ResultType>::value;*/\
+        static const size_t s_numResults = ScriptCanvas::Internal::extended_tuple_size<ResultType>::value;\
         \
         static AZStd::string GetArgName(size_t i)\
         {\
+            if (s_numArgs < 2)\
```
Comment (onecent1101) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/NodeFunctionGeneric.h:87]: please help me to refresh memory, 2 means including execution in and out slot?

**Review by onecent1101 — APPROVED** [review]

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/NodeFunctionGeneric.h (hunk 1 above)
Comment (carlitosan) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/NodeFunctionGeneric.h]: in this case its the number of data inputs on the Node, which correspond to the number of arguments to the C++ function.

Comment (onecent1101) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/NodeFunctionGeneric.h]: oh yes it is for c++ function not scriptcanvas node.
btw, just to confirm, argument list won't have this pointer as first arg, right? I remember I saw this kind of thing somewhere before.

**Review by AMZN-Gene — APPROVED** [review]

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/NodeFunctionGeneric.h (hunk 1 above)
Comment (carlitosan) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/NodeFunctionGeneric.h]: Correct, that's for behavior context method nodes written against class methods.

**Review by cgalvan — APPROVED** [review]

**Pull request merged**